pynecone/utils.py

```diff
@@ -799,41 +794,79 @@ def get_path_args(path: str) -> Dict[str, str]:
     Args:
         path: The path to get the arguments for.
 
-    Raises:
-        ValueError: explains what is wrong with the path.
-
     Returns:
         The path arguments.
     """
-    # Import here to avoid circular imports.
-    from pynecone.var import BaseVar
+
+    def add_path_arg(match, type_):
+        """Add arg from regex search result.
+
+        Args:
+            match: result of a regex search
+            type_: the assigned type for this arg
+
+        Raises:
+            ValueError: explains what is wrong with the path.
+        """
+        arg_name = match.groups()[0]
+        if arg_name in args:
+            raise ValueError(
+                f"arg name [{arg_name}] is used more than once in this URL"
+            )
+        args[arg_name] = type_
 
     # Regex to check for path args.
-    check = re.compile(r"^\[(.+)\]$")
-    check_catchall = re.compile(r"^\[\.\.\.(.+)\]$")
+    check = re.compile(constants.Regex.ARG)
+    check_strict_catchall = re.compile(constants.Regex.STRICT_CATCHALL)
+    check_opt_catchall = re.compile(constants.Regex.OPT_CATCHALL)
 
     # Iterate over the path parts and check for path args.
     args = {}
     for part in path.split("/"):
-        match = check_catchall.match(part)
-        if match:
-            arg_name = match.groups()[0]
-            if arg_name in args:
-                raise ValueError(f"arg [{arg_name}] is used more than once in this URL")
+        match_opt = check_opt_catchall.match(part)
+        if match_opt:
+            add_path_arg(match_opt, "catchall")
+            break
 
-            args[arg_name] = "catchall"
-            continue
+        match_strict = check_strict_catchall.match(part)
+        if match_strict:
+            add_path_arg(match_strict, "catchall")
+            break
 
         match = check.match(part)
         if match:
             # Add the path arg to the list.
-            arg_name = match.groups()[0]
-            if arg_name in args:
-                raise ValueError(f"arg [{arg_name}] is used more than once in this URL")
-            args[arg_name] = "patharg"
+            add_path_arg(match, "patharg")
     return args
 
 
+def catchall_in_route(route) -> str:
```

Comment:
Missing arg type annotation (I'll try to make the linter stricter on this to catch earlier)